pkg/daemon/server/service/pipeline_watermark_query.go

```diff
@@ -113,3 +113,56 @@ func (ps *pipelineMetadataQuery) GetVertexWatermark(ctx context.Context, request
 	resp.VertexWatermark = v
 	return resp, nil
 }
+
+// GetPipelineWatermark is used to return the head watermark for a given pipeline.
+func (ps *pipelineMetadataQuery) GetPipelineWatermark(ctx context.Context, request *daemon.GetPipelineWatermarkRequest) (*daemon.GetPipelineWatermarkResponse, error) {
+	resp := new(daemon.GetPipelineWatermarkResponse)
+	retFalse := false
+	retTrue := true
+
+	// If watermark is not enabled, return time zero
+	if ps.pipeline.Spec.Watermark.Disabled {
+		timeZero := time.Unix(0, 0).UnixMilli()
+		watermarkArr := make([]*daemon.VertexWatermark, len(ps.watermarkFetchers))
+		i := 0
```

Comment:
What is `i` used to do? It looks like it's always 0.

Reply:
forgot the incremental operation for `i`. added it now